autonomy/analyse/abci/app_spec.py

```diff
@@ -338,6 +343,51 @@ def abci_to_dfa(cls, abci_app_cls: Any, label: str = "") -> "DFA":
         )
 
 
+event_pattern = re.compile(r"Event\.(\w+)", re.DOTALL)
+
+
+def _check_unreferenced_events(abci_app_cls: AbciApp) -> None:
```

Comment:
Why keep this out of the method?

Reply:
To avoid compiling the regular expression (as the method is called multiple times).

Reply:
Ok, I updated it in ee862018b729dae611a4842f16091339d97df304.